dali/operators/image/resize/resize.h

```diff
@@ -35,42 +36,90 @@ namespace detail {
     const TransformMeta &meta, kernels::FilterDesc min_filter, kernels::FilterDesc mag_filter);
 }  // namespace detail
 
-class ResizeAttr : protected ResizeCropMirrorAttr {
- public:
-  explicit inline ResizeAttr(const OpSpec &spec) : ResizeCropMirrorAttr(spec) {}
-
-  void SetBatchSize(int batch_size) {
-    per_sample_meta_.reserve(batch_size);
-  }
-
- protected:
-  uint32_t ResizeInfoNeeded() const override { return 0; }
-
-  // store per-thread data for same resize on multiple data
-  std::vector<TransformMeta> per_sample_meta_;
-};
-
 template <typename Backend>
 class Resize : public Operator<Backend>
-             , protected ResizeAttr
-             , protected ResizeBase {
+             , protected ResizeBase<Backend> {
  public:
   explicit Resize(const OpSpec &spec);
 
  protected:
-  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
-    return false;
+  int NumSpatialDims() const { return resize_attr_.spatial_ndim_; }
+  int FirstSpatialDim() const { return resize_attr_.first_spatial_dim_; }
+
+  bool CanInferOutputs() const override { return true; }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override;
+
+  void RunImpl(workspace_t<Backend> &ws) override;
+
+  void SaveAttrs(const TensorListView<StorageCPU, int, 1> &shape_data,
+                 const TensorListShape<> &orig_shape) const {
+    int N = orig_shape.num_samples();
+    for (int i = 0; i < N; i++) {
+      auto sample_shape = orig_shape.tensor_shape_span(i);
+      int *out_shape = shape_data.data[i];
+      for (int d = 0; d < NumSpatialDims(); d++) {
+        out_shape[d] = sample_shape[FirstSpatialDim() + d];
+      }
+    }
+  }
+
+  void PrepareParams(const ArgumentWorkspace &ws, const TensorListShape<> &input_shape,
+                     const TensorLayout &layout) {
+    resize_attr_.PrepareResizeParams(spec_, ws, input_shape, layout);
+    assert(NumSpatialDims() >= 1 && NumSpatialDims() <= 3);
```

Comment:
We don't support NumSpatialDims == 3 just yet, right?

Reply:
It's enforced elsewhere I think.